src/groupeddataframe/splitapplycombine.jl

```diff
@@ -818,9 +820,12 @@ function _combine_with_first(first::Union{NamedTuple, DataFrameRow, AbstractData
     let eltys=eltys, n=n # Workaround for julia#15276
         initialcols = ntuple(i -> Tables.allocatecolumn(eltys[i], n), _ncol(first))
     end
-    outcols = _combine_with_first!(first, initialcols, idx, 1, 1, f, gd, incols,
-                                   tuple(propertynames(first)...))
-    idx, outcols, propertynames(first)
+    targetcolnames = tuple(propertynames(first)...)
+    outcols, finalcolnames = first isa Union{AbstractDataFrame,
+                              NamedTuple{<:Any, <:Tuple{Vararg{AbstractVector}}}} ?
+              _combine_with_first!(first, initialcols, idx, 1, 1, f, gd, incols, targetcolnames) :
```

Comment:
Maybe call this `_combine_with_first_table!`? Or even `_combine_tables_with_first!` and `_combine_rows_with_first!`?

BTW, it would be nice to file an issue in Julia with a branch to reproduce the bug (you can see in the file that I had similar issues in older Julia versions).

Reply:
Fixed. I will try to recreate the dispatch problem and file an issue.